lib/client.js

```diff
@@ -901,6 +905,16 @@ function onSocketConnect () {
 }
 
 function onSocketError (err) {
+  const { [kParser]: parser } = this
+  if (err.code === 'ECONNRESET' && parser.statusCode && !parser.shouldKeepAlive) {
```

Comment:
Do we need this here? Isn't it enough to apply this logic on "graceful" socket shutdown?

Reply:
It's really OS-dependent. We get an ECONNRESET on Mac and a normal shutdown on Linux.